src/raft.c

```diff
@@ -341,23 +361,43 @@ void RaftExecuteCommandArray(RedisRaftCtx *rr,
         int old_entered_eval = rr->entered_eval;
 
         if ((cmdlen == 4 && !strncasecmp(cmd, "eval", 4)) ||
-            (cmdlen == 7 && !strncasecmp(cmd, "evalsha", 7))) {
+            (cmdlen == 7 && !strncasecmp(cmd, "evalsha", 7)) ||
+            (cmdlen == 5 && !strncasecmp(cmd, "fcall", 5))) {
             rr->entered_eval = 1;
         }
 
+        char *resp_call_fmt;
+
+        if (reply_ctx) {
+            if (cmds->acl) {
+                /* We have client so check ACL (C), return response in client format (0) and rm_call error as reply (E) */
+                resp_call_fmt = "CE0v";
+            } else {
+                resp_call_fmt = "0v";
+            }
+        } else {
+            if (cmds->acl) {
+                /* We don't have a client so only check ACL (C) */
+                resp_call_fmt = "Cv";
+            } else {
+                resp_call_fmt = "v";
+            }
+        }
+
         enterRedisModuleCall();
-        reply = RedisModule_Call(ctx, cmd,
-                                 rr->resp_call_fmt, &c->argv[1], c->argc - 1);
-        int ret_errno = errno;
+        RedisModule_SetContextUser(ctx, user);
+        reply = RedisModule_Call(ctx, cmd, resp_call_fmt, &c->argv[1], c->argc - 1);
         exitRedisModuleCall();
 
         rr->entered_eval = old_entered_eval;
 
         if (reply_ctx) {
-            if (reply) {
-                RedisModule_ReplyWithCallReply(reply_ctx, reply);
+            int type = RedisModule_CallReplyType(reply);
+            if (type != REDISMODULE_REPLY_ERROR) {
+                RedisModule_ReplyWithCallReply(ctx, reply);
             } else {
-                replyRMCallError(reply_ctx, ret_errno, cmd, cmdlen);
+                const char *str = RedisModule_CallReplyStringPtr(reply, NULL);
+                RedisModule_ReplyWithError(ctx, str);
```

Comment:
Why do we need these if checks? If we go with `CE0v` flags, can we just do:

```c


if (reply_ctx) {
    RedisModule_ReplyWithCallReply(reply_ctx, reply);
}

RedisModule_FreeCallReply(reply);
```

Reply:
Really? It should not crash, sounds like a bug in Redis. I thought the whole purpose of  `E` flag is being able to reply to the client without doing any additional checks.

Can you check it again with a clean build? I just make this change and it doesn't crash for me:

```c 
reply = RedisModule_Call(ctx, "some_non_existing_command", "E0v", &c->argv[1], c->argc - 1);
RedisModule_ReplyWithCallReply(reply_ctx, reply);
```

Reply:
I ran into the problem with my dry run tests and the module, I'll try again.  If the above is working, perhaps was a temporary condition that was fixed (and yes, I agree, I'd expect it to just work).

Reply:
ok, made it like this, seems to be passing now, unsure why I had issues before.